File: uhrwerk-metastore/src/main/java/io/qimia/uhrwerk/common/metastore/config/ConnectionResult.java
package io.qimia.uhrwerk.common.metastore.config;

import io.qimia.uhrwerk.common.metastore.model.ConnectionModel;

import java.io.Serializable;
import java.util.Objects;

public class ConnectionResult implements Serializable {
  private static final long serialVersionUID = 3800840086237687091L;
  ConnectionModel newConnection;
  ConnectionModel oldConnection;
  boolean success;
  boolean error;
  String message;
  Exception exception;

  public ConnectionModel getNewConnection() {
    return newConnection;
  }

  public void setNewConnection(ConnectionModel newConnection) {
    this.newConnection = newConnection;
  }

  public ConnectionModel getOldConnection() {
    return oldConnection;
  }

  public void setOldConnection(ConnectionModel oldConnection) {
    this.oldConnection = oldConnection;
  }

  public boolean isSuccess() {
    return success;
  }

  public void setSuccess(boolean success) {
    this.success = success;
  }

  public boolean isError() {
    return error;
  }

  public void setError(boolean error) {
    this.error = error;
  }

  public String getMessage() {
    return message;
  }

  public void setMessage(String message) {
    this.message = message;
  }

  public Exception getException() {
    return exception;
  }

  public void setException(Exception exception) {
    this.exception = exception;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    ConnectionResult that = (ConnectionResult) o;
    return success == that.success
        && error == that.error
        && Objects.equals(newConnection, that.newConnection)
        && Objects.equals(oldConnection, that.oldConnection)
        && Objects.equals(message, that.message)
        && Objects.equals(exception, that.exception);
  }

  @Override
  public int hashCode() {
    return Objects.hash(newConnection, oldConnection, success, error, message, exception);
  }

  @Override
  public String toString() {
    return "ConnectionResult{"
        + "newConnection="
        + newConnection
        + ", oldConnection="
        + oldConnection
        + ", success="
        + success
        + ", error="
        + error
        + ", message='"
        + message
        + '\''
        + ", exception="
        + exception
        + '}';
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/config/ConnectionResult.kt
package io.qimia.uhrwerk.common.metastore.config

import io.qimia.uhrwerk.common.metastore.model.ConnectionModel
import java.io.Serializable
import java.util.*

class ConnectionResult : Serializable {
    var newConnection: ConnectionModel? = null
    var oldConnection: ConnectionModel? = null
    var isSuccess = false
    var isError = false
    var message: String? = null
    var exception: Exception? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ConnectionResult
        return isSuccess == that.isSuccess && isError == that.isError && newConnection == that.newConnection && oldConnection == that.oldConnection && message == that.message && exception == that.exception
    }

    override fun hashCode(): Int {
        return Objects.hash(newConnection, oldConnection, isSuccess, isError, message, exception)
    }

    override fun toString(): String {
        return ("ConnectionResult{"
                + "newConnection="
                + newConnection
                + ", oldConnection="
                + oldConnection
                + ", success="
                + isSuccess
                + ", error="
                + isError
                + ", message='"
                + message
                + '\''
                + ", exception="
                + exception
                + '}')
    }

}